File: LuaScripts/Public/NodeGraph/Nodes/AddMonsterNode.lua

function SvrHandler(tNodeGraph, tNodeData)
    local tSvrGame = SvrGameMod.getGameById(tNodeGraph.nGameId)
    local nRefreshId = tNodeData.nRefreshId
    if nRefreshId and nRefreshId ~= "" then
        SvrGameRoleMod.refreshMonster(tSvrGame, nRefreshId)
    else
        local nConfigId = tNodeData.nConfigId
        local nPosX, nPosY, nPosZ = 0, 0, 0
        local nPosId = tNodeData.nPosId
        if nPosId ~= nil and nPosId ~= "" then
            local tPos = GameSceneCfgMod.getPosition(tSvrGame.tGameSceneConfig, nPosId)
            if tPos ~= nil then
                nPosX, nPosY, nPosZ = tPos.x, tPos.y, tPos.z
            end
        else
            nPosX, nPosY, nPosZ = tNodeData.nPosX, tNodeData.nPosY, tNodeData.nPosZ
        end
        SvrGameRoleMod.addMonster(tSvrGame, nConfigId, nPosX, nPosY, nPosZ)
    end

    SvrNodeGraphMod.finishNode(tNodeGraph, tNodeData.nNodeId, true)
end

File: Assets/Scripts/Editor/SceneNodeGraph/Nodes/AddMonsterNode.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace SceneNodeGraph
{
    public class AddMonsterNode : BaseNode
    {
        public int nRefreshId;
        [RequireInt("nRefreshId", 0)]
        public int nConfigId;
        [RequireInt("nRefreshId", 0)]
        public int nPosId;
        [RequireInt("nPosId", 0)]
        public float nPosX;
        [RequireInt("nPosId", 0)]
        public float nPosY;
        [RequireInt("nPosId", 0)]
        public float nPosZ;
    }

}
